src/Microsoft.AspNet.Server.Testing/Deployers/ApplicationDeployer.cs

```diff
@@ -268,5 +270,31 @@ protected static string GetOSPrefix()
                     throw new InvalidOperationException("Unrecognized operating system");
             }
         }
+
+        protected Uri FindFreeUrl(string urlHint)
+        {
+            var uriHint = new Uri(urlHint);
+            var builder = new UriBuilder(uriHint)
+            {
+                Port = FindFreePort(uriHint.Port)
+            };
+            return builder.Uri;
+        }
+
+        protected static int FindFreePort(int initialPort = 0)
```

Comment:
What about just move it to a helper class instead of a protected method?

And we are encouraged not to use default parameter value. Instead use methods overload.